myapp/templates/index.html

```diff
@@ -29,6 +32,20 @@
                     text-danger
                 {% endif %}
                 ">{{ stock.change_percent|floatformat:2 }}%</td>
+                {% if request.user.is_authenticated %}

```

Comment:
maybe here you can check if the profile is set or not; you already checked if authenticated in the view.
```suggestion
                {% if profile %}

```

Reply:
You are right, I will test this and if there are no problems, I will commit this change

Reply:
I am now using `{% if profile %}` everywhere in the templates (expect base.html)